libsession/src/main/java/org/session/libsession/messaging/sending_receiving/pollers/OpenGroupPoller.kt

```diff
@@ -39,15 +40,97 @@ class OpenGroupPoller(private val server: String, private val executorService: S
     var isCaughtUp = false
     var secondToLastJob: MessageReceiveJob? = null
     private var future: ScheduledFuture<*>? = null
+    private var runId: UUID = UUID.randomUUID()
 
     companion object {
         private const val pollInterval: Long = 4000L
         const val maxInactivityPeriod = 14 * 24 * 60 * 60 * 1000
+
+        public fun handleRoomPollInfo(
+            server: String,
+            roomToken: String,
+            pollInfo: OpenGroupApi.RoomPollInfo,
+            createGroupIfMissingWithPublicKey: String? = null
+        ) {
+            val storage = MessagingModuleConfiguration.shared.storage
+            val groupId = "$server.$roomToken"
+            val dbGroupId = GroupUtil.getEncodedOpenGroupID(groupId.toByteArray())
+            val existingOpenGroup = storage.getOpenGroup(roomToken, server)
+
+            // If we don't have an existing group and don't have a 'createGroupIfMissingWithPublicKey'
+            // value then don't process the poll info
+            val publicKey = ((existingOpenGroup?.publicKey ?: createGroupIfMissingWithPublicKey) ?: return)
+
+            val openGroup = OpenGroup(
+                server = server,
+                room = pollInfo.token,
+                name = ((pollInfo.details?.name ?: existingOpenGroup?.name) ?: ""),
+                publicKey = publicKey,
+                imageId = (pollInfo.details?.imageId ?: existingOpenGroup?.imageId),
+                canWrite = pollInfo.write,
+                infoUpdates = ((pollInfo.details?.infoUpdates ?: existingOpenGroup?.infoUpdates) ?: 0)
```

Comment:
looks better! there's just unnecessary parentheses x3

Reply:
Same as the above - the unnecessary parentheses are for readability due to the double elvis operator